pkgs/applications/science/misc/colmap/default.nix

```diff
@@ -1,10 +1,14 @@
 { mkDerivation, lib, fetchFromGitHub, cmake, boost17x, ceres-solver, eigen,
   freeimage, glog, libGLU, glew, qtbase,
-  cudaSupport ? false, cudatoolkit ? null }:
+  cudaSupport ? false, cudaPackages }:
 
-assert !cudaSupport || cudatoolkit != null;
+assert cudaSupport -> cudaPackages != { };
 
-let boost_static = boost17x.override { enableStatic = true; };
+let
+  boost_static = boost17x.override { enableStatic = true; };
```

Comment:
Why do we need `enableStatic`? Why wasn't this necessary before?

Reply:
oh oops just saw that this was necessary before... wait why is it necessary? does clomp support building against shared boost libs?

Reply:
I had the same questions, but decided not to touch it for now. I don't know why it uses static. Although static is by default preferable to dynamic. I also don't know why it pins a narrow range for boost versions

Reply:
> Although static is by default preferable to dynamic.

Why is this? In Nix, shared libraries seem much more space efficient while still locking down the exact library versions. What's the benefit to static libraries?

Reply:
Right, I guess "by default" is plain incorrect wording. There is "a perspective", from which static builds are still "preferable": you get _fewer_ derivations in the runtime closure, might even get binaries that you can patchelf and ship to a non-nixos system, might even get a smaller runtime closure in some corner cases, when the expression shipping shared libraries also propagates irrelevant. For example, our `symlinkJoin`-ed cuda is a runtime dependency that ships shared libraries, but also nvcc and static archives: nix does not have means to know it won't need them at runtime. If all of cuda was linked statically, that wouldn't happen

Reply:
I have already forgotten the reason for this decision. It seems the comap build system required static boost

Reply:
Yeah, I'd be curious to track that down. But that's out of scope for this PR